tokenserver-db/src/models.rs

```diff
@@ -685,156 +684,220 @@ impl TokenserverDb {
+    }
+
+    async fn get_service_id(
+        &mut self,
+        params: params::GetServiceId,
+    ) -> Result<results::GetServiceId, DbError> {
+        TokenserverDb::get_service_id(self, params).await
+    }
 
     fn timeout(&self) -> Option<Duration> {
         self.timeout
     }
 
     #[cfg(test)]
-    async_db_method!(get_user, TokenserverDb::get_user, GetUser);
+    async fn get_user(&mut self, params: params::GetUser) -> Result<results::GetUser, DbError> {
+        TokenserverDb::get_user(self, params).await
+    }
 
-    fn check(&mut self) -> DbFuture<'_, results::Check, DbError> {
-        Box::pin(TokenserverDb::check(self))
+    async fn check(&mut self) -> Result<results::Check, DbError> {
+        TokenserverDb::check(self).await
     }
 
     #[cfg(test)]
-    async_db_method!(
-        set_user_created_at,
-        TokenserverDb::set_user_created_at,
-        SetUserCreatedAt
-    );
+    async fn set_user_created_at(
+        &mut self,
+        params: params::SetUserCreatedAt,
+    ) -> Result<results::SetUserCreatedAt, DbError> {
+        TokenserverDb::set_user_created_at(self, params).await
+    }
 
     #[cfg(test)]
-    async_db_method!(
-        set_user_replaced_at,
-        TokenserverDb::set_user_replaced_at,
-        SetUserReplacedAt
-    );
+    async fn set_user_replaced_at(
+        &mut self,
+        params: params::SetUserReplacedAt,
+    ) -> Result<results::SetUserReplacedAt, DbError> {
+        TokenserverDb::set_user_replaced_at(self, params).await
+    }
 
     #[cfg(test)]
-    async_db_method!(post_node, TokenserverDb::post_node, PostNode);
+    async fn post_node(&mut self, params: params::PostNode) -> Result<results::PostNode, DbError> {
+        TokenserverDb::post_node(self, params).await
+    }
 
     #[cfg(test)]
-    async_db_method!(get_node, TokenserverDb::get_node, GetNode);
+    async fn get_node(&mut self, params: params::GetNode) -> Result<results::GetNode, DbError> {
+        TokenserverDb::get_node(self, params).await
+    }
 
     #[cfg(test)]
-    async_db_method!(unassign_node, TokenserverDb::unassign_node, UnassignNode);
+    async fn unassign_node(
+        &mut self,
+        params: params::UnassignNode,
+    ) -> Result<results::UnassignNode, DbError> {
+        TokenserverDb::unassign_node(self, params).await
+    }
 
     #[cfg(test)]
-    async_db_method!(remove_node, TokenserverDb::remove_node, RemoveNode);
+    async fn remove_node(
+        &mut self,
+        params: params::RemoveNode,
+    ) -> Result<results::RemoveNode, DbError> {
+        TokenserverDb::remove_node(self, params).await
+    }
 
     #[cfg(test)]
-    async_db_method!(post_service, TokenserverDb::post_service, PostService);
+    async fn post_service(
+        &mut self,
+        params: params::PostService,
+    ) -> Result<results::PostService, DbError> {
+        TokenserverDb::post_service(self, params).await
+    }
 }
 
+#[async_trait(?Send)]
```

Comment:
This was that weird negation macro thing you mentioned, right? So the compiler doesn't break if the trait bound is not satisfied?